src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -245,43 +248,79 @@ private static void ApplyProperties(ISupportProperties telemetry, IEnumerable<Ke
                 {
                     case LogConstants.NameKey:
                         functionName = value.Value.ToString();
                         break;
+                    case LogConstants.TotalDurationKey:
+                        totalDuration = ConvertToDouble(value.Value);
+                        break;
+                    case LogConstants.MaxDurationKey:
+                        maxDuration = ConvertToDouble(value.Value);
+                        break;
+                    case LogConstants.MinDurationKey:
+                        minDuration = ConvertToDouble(value.Value);
+                        break;
+                    case LogConstants.SuccessesKey:
+                        successCount = ConvertToInt(value.Value);
+                        break;
+                    case LogConstants.FailuresKey:
+                        failureCount = ConvertToInt(value.Value);
+                        break;
                     case LogConstants.TimestampKey:
                     case LogConstants.OriginalFormatKey:
                         // Timestamp is created automatically
                         // We won't use the format string here
                         break;
                     default:
-                        if (value.Value is TimeSpan)
-                        {
-                            // if it's a TimeSpan, log the milliseconds
-                            metrics.Add(value.Key, ((TimeSpan)value.Value).TotalMilliseconds);
-                        }
-                        else if (value.Value is double || value.Value is int)
-                        {
-                            metrics.Add(value.Key, Convert.ToDouble(value.Value));
-                        }
-
                         // do nothing otherwise
                         break;
                 }
             }
 
-            foreach (KeyValuePair<string, double> metric in metrics)
+            int count = failureCount + successCount;
+            double successRate = Math.Round((successCount / (double)count) * 100, 2);
+
+            // Metric names will be created like "{FunctionName} {MetricName}"
+            _telemetryClient.TrackMetric(new MetricTelemetry($"{functionName} {LogConstants.DurationKey}", totalDuration) { Count = count, Min = minDuration, Max = maxDuration });
+            _telemetryClient.TrackMetric(new MetricTelemetry($"{functionName} {CountMetricName}", count));
+            _telemetryClient.TrackMetric(new MetricTelemetry($"{functionName} {LogConstants.SuccessesKey}", successCount));
+            _telemetryClient.TrackMetric(new MetricTelemetry($"{functionName} {LogConstants.FailuresKey}", failureCount));
+            _telemetryClient.TrackMetric(new MetricTelemetry($"{functionName} {SuccessRateMetricName}", successRate));
+        }
+
+        private static double ConvertToDouble(object value)
+        {
+            double convertedValue = 0;
+
+            if (value is int || value is double)
```

Comment:
do we want to allow this for all types, i.e. a string "120"

Could wrap the whole convert switch above in a try catch to handle formatting.

Reply:
For now, we own this "contract" so we know what data to expect. There's no real need to support strings -- we could eventually though.

Reply:
In that case, why not directly use Convert.ToDouble?

Reply:
Being overly-defensive, I guess. I could -- it wouldn't break anything.